src/popup.ts

```diff
@@ -90,60 +97,43 @@ const ALLOWED_URL_PATTERNS: Array<RegExp | string> = [
 async function setupPopup() {
   const urlInput = /** @type {HTMLInputElement} */ document.getElementById(
     'url-input'
-  ) as HTMLInputElement;
+  ) as MdFilledTextField;
   const form = document.getElementsByTagName('form')[0];
-  const podficLabel = document.getElementById(
-    'podfic_label'
-  ) as HTMLInputElement;
+  const podficLabel = document.getElementById('podfic_label') as MdCheckbox;
   const podficLengthLabel = document.getElementById(
     'podfic_length_label'
-  ) as HTMLInputElement;
+  ) as MdCheckbox;
   const podficLengthValue = document.getElementById(
     'podfic_length_value'
-  ) as HTMLInputElement;
+  ) as MdFilledSelect;
   const titleFormatValue =
     /** @type {HTMLInputElement} */ document.getElementById(
       'title_template_value'
-    ) as HTMLInputElement;
+    ) as MdFilledSelect;
   const summaryFormatValue = document.getElementById(
     'summary_template_value'
-  ) as HTMLInputElement;
-  const urlTextField = new MDCTextField(
-    document.querySelector('.mdc-text-field')!
-  );
+  ) as MdFilledSelect;
   const snackbar = new MDCSnackbar(document.querySelector('.mdc-snackbar')!);
   const submitButton = document.querySelector('#import') as HTMLButtonElement;
   const optionsLink = document.getElementById(
     'options-link'
   ) as HTMLAnchorElement;
   optionsLink.href = browser.runtime.getURL('options.html');
 
-  // Setting this means that we have to update the validity of the text field
-  // when native validation shows the field as invalid. This is the only way
-  // we can keep validation in sync with our submit only validity checks.
-  urlTextField.useNativeValidation = false;
-
   // Defensively, we add the listeners first, so even if we fail to read some
   // information from storage we should be able to recover on submit.
 
-  urlInput.addEventListener('input', () => {
-    // Always clear the custom error when the user changes the value.
-    urlTextField.helperTextContent = '';
-    // Keep the text field in sync with the input.
-    urlTextField.valid = urlInput.validity.valid;
-  });
-
```

Comment:
Did this stop being necessary because the native validation is smarter somehow now?

Reply:
Yeah whatever quirk made this necessary is no longer necessary